backend/typescript/migrations/20220615224128-change-tag-from-review-to-book.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable */
+"use strict";
+
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    try {
+      await queryInterface.dropTable("review_tag");
+      await queryInterface.dropTable("tags");
+      await queryInterface.createTable(
+        'tags',
+        {
+          name: {
+            type: Sequelize.STRING,
+            primaryKey: true,
+          },
+          createdAt: {
+            type: Sequelize.DATE
+          },
+          updatedAt: {
+            type: Sequelize.DATE
+          },
+        },
+      );
+      await queryInterface.createTable(
+        'book_tag',
+        {
+          book_id: {
+            type: Sequelize.INTEGER,
+            primaryKey: true,
+            references: {
+              model: 'books',
+              key: 'id'
+            },
+          },
+          tag_name: {
+            type: Sequelize.STRING,
+            primaryKey: true,
+            references: {
+              model: 'tags',
+              key: 'name'
+            },
+          },
+          createdAt: {
+            type: Sequelize.DATE
+          },
+          updatedAt: {
+            type: Sequelize.DATE
+          },
+        },
+      );
+        return Promise.resolve();
+
+      } catch (e) {
+        return Promise.reject(e);
+        
+      }
+  },
+
+  async down(queryInterface, Sequelize) {
+    try {
+      await queryInterface.dropTable("book_tag");
+      await queryInterface.dropTable("tags");
+      await queryInterface.createTable(
+        'tags',
+        {
+          id: {
+            type: Sequelize.INTEGER,
+            primaryKey: true,
+          },
+          name: {
+            type: Sequelize.STRING,
+          },
+          createdAt: {
+            type: Sequelize.DATE
+          },
+          updatedAt: {
+            type: Sequelize.DATE
+          },
+        },
+      );
+      await queryInterface.createTable(
+        'review_tag',
```

Comment:
Just wondering, what would be the point of doing this? After dropping it when we migrate up and then try to undo the migration, wouldn't the data within the table already be gone? So why bother creating the table again? @susheelk thoughts?

Reply:
also we would have already deleted the review_tag model in the backend so backend and db would not sync up either

Reply:
Yeah, this doesn't preserve data (which is fine for this case) @susheelk can you weigh in this?

Reply:
This is technically correct, but when writing migrations, our down should **always** be an exact reverse of our up. Basically we would only expect `down` to work on an older commit (ie before this PR) when tags were associated with reviews